hcloud/core/client.py

```diff
@@ -86,11 +60,8 @@ class GetEntityByNameMixin:
 
     def get_by_name(self, name):
         # type: (str) -> BoundModelBase
-        self._is_list_attribute_implemented()
-        response = self.get_list(name=name)
-        entities = getattr(response, self.results_list_attribute_name)
-        entity = entities[0] if entities else None
-        return entity
+        entities, _ = self.get_list(name=name)
```

Comment:
`ClientEntityBase.add_meta_to_result` previously guaranteed that the right attribute from the tuple was picked. Now that we have x `ResourcePageResult` classes, someone might implement one where the attributes are ordered the other way round, breaking this and the `_get_all` methods.

Is there some way we can force all `XXPageResult` to have the same order?

Reply:
I think it's too much hassle to enforce it. We can rely on the tests for get_list and get_all to fail if someone adds an extra attribute (3 elements) or put them in the wrong order (meta, result).

We could put an assertion within the _get_list method to make sure we have the right type and provide a clearer explanation to the dev. But I think the errors will be self-explanatory.